sharding-core/sharding-core-optimize/src/main/java/org/apache/shardingsphere/core/optimize/sharding/segment/select/item/SelectItems.java

```diff
@@ -46,6 +49,8 @@
     private final boolean distinctRow;
     
     private final Collection<SelectItem> items;
+
+    private final Map<String, TableMetaData> ownerTableMetaDataMap;
```

Comment:
Can this class not hold table metadata, instead of fetch it from table metadata with table name?